### PR TITLE
Fixed issue 47, change the template name from "spring" to "khonsu".

### PR DESCRIPTION
**Pull request by ahamed**
Pull Request for Issue # .
This PR fixes the issue #47 which is changes the template name from "spring" to "khonsu".

### CODE REVIEW

File: administrator/components/com_config/tmpl/application/default.php [hunk 1]
```diff
@@ -31,9 +31,9 @@
 <form action="<?php echo Route::_('index.php?option=com_config'); ?>" id="application-form" method="post" name="adminForm" class="d-flex form-validate" data-cancel-task="config.cancel.component">
 	<!-- Begin Sidebar -->
 	<div id="sidebar" class="com-config-sidebar">
-		<button class="btn btn-sm btn-secondary my-2 options-menu d-md-none" type="button" data-toggle="collapse" data-target=".sidebar-nav" aria-controls="sidebar-nav" aria-expanded="false" aria-label="<?php echo Text::_('TPL_SPRING_TOGGLE_SIDEBAR'); ?>">
+		<button class="btn btn-sm btn-secondary my-2 options-menu d-md-none" type="button" data-toggle="collapse" data-target=".sidebar-nav" aria-controls="sidebar-nav" aria-expanded="false" aria-label="<?php echo Text::_('TPL_KHONSU_TOGGLE_SIDEBAR'); ?>">
```
Comment (brianteeman) [inline administrator/components/com_config/tmpl/application/default.php]: You should not use a template specific language string in a core non template file. See https://github.com/joomla/joomla-cms/pull/27082 for reference

File: administrator/components/com_config/tmpl/component/default.php [hunk 2]
```diff
@@ -43,9 +43,9 @@
 <form action="<?php echo Route::_('index.php?option=com_config'); ?>" id="component-form" method="post" class="form-validate" name="adminForm" autocomplete="off" data-cancel-task="config.cancel.component">
 		<?php // Begin Sidebar ?>
 		<div class="com-config-sidebar" id="sidebar">
-			<button class="btn btn-sm btn-secondary my-2 options-menu d-md-none" type="button" data-toggle="collapse" data-target=".sidebar-nav" aria-controls="sidebar-nav" aria-expanded="false" aria-label="<?php echo Text::_('TPL_SPRING_TOGGLE_SIDEBAR'); ?>">
+			<button class="btn btn-sm btn-secondary my-2 options-menu d-md-none" type="button" data-toggle="collapse" data-target=".sidebar-nav" aria-controls="sidebar-nav" aria-expanded="false" aria-label="<?php echo Text::_('TPL_KHONSU_TOGGLE_SIDEBAR'); ?>">
```
Comment (brianteeman) [inline administrator/components/com_config/tmpl/component/default.php]: You should not use a template specific language string in a core non template file. See https://github.com/joomla/joomla-cms/pull/27082 for reference

File: administrator/templates/khonsu/joomla.asset.json [hunk 3]
```diff
@@ -1,12 +1,12 @@
 {
   "$schema": "https://developer.joomla.org/schemas/json-schema/web_assets.json",
-  "name": "spring",
+  "name": "khonsu",
   "version": "4.0.0",
-  "description": "Spring is the Joomla 4 administrator template",
+  "description": "Khonsu is the Joomla 4 administrator template",
```
Comment (brianteeman) [inline administrator/templates/khonsu/joomla.asset.json]:
```suggestion
  "description": "Khonsu is a Joomla 4 administrator template",
```

File: administrator/templates/khonsu/language/en-GB/en-GB.tpl_khonsu.ini [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+TPL_KHONSU_IMAGE_SETTINGS_LABEL="Image Settings"
+TPL_KHONSU_LOGIN_LOGO_LABEL="Login Logo"
+TPL_KHONSU_LOGIN_SIDEBAR_VIEW_WEBSITE="view website"
+TPL_KHONSU_LOGIN_SITE_TITLE="%s - Administrator Login"
+TPL_KHONSU_SITENAME_LABEL="Site Name"
+TPL_KHONSU_SITE_LOGO_LABEL="Site Logo"
+TPL_KHONSU_SITE_LOGO_SMALL_LABEL="Small Site Logo"
+TPL_KHONSU_TOGGLE_SIDEBAR="Toggle Menu"
+TPL_KHONSU_TOOLBAR="Toolbar"
+TPL_KHONSU_XML_DESCRIPTION="Continuing the Egyptian god/goddess theme (Khepri from 1.5, Hathor from 1.6, Isis from 3.0), Khonsu is the name of the Joomla 4 administrator template."
```
Comment (brianteeman) [inline administrator/templates/khonsu/language/en-GB/en-GB.tpl_khonsu.ini]:
```suggestion
TPL_KHONSU_XML_DESCRIPTION="Continuing the Egyptian god/goddess theme (Khepri from 1.5, Hathor from 1.6, Isis from 3.0), Khonsu is the name of a Joomla 4 administrator template."
```

File: administrator/templates/khonsu/language/en-GB/en-GB.tpl_khonsu.sys.ini [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+TPL_KHONSU_POSITION_BOTTOM="Bottom"
+TPL_KHONSU_POSITION_CPANEL="Cpanel"
+TPL_KHONSU_POSITION_DEBUG="Debug"
+TPL_KHONSU_POSITION_ICON="Quick Icons"
+TPL_KHONSU_POSITION_LOGIN="Login"
+TPL_KHONSU_POSITION_MENU="Menu"
+TPL_KHONSU_POSITION_STATUS="Status"
+TPL_KHONSU_POSITION_TITLE="Title"
+TPL_KHONSU_POSITION_TOOLBAR="Toolbar"
+TPL_KHONSU_XML_DESCRIPTION="Continuing the Egyptian god/goddess theme (Khepri from 1.5, Hathor from 1.6, Isis from 3.0), Khonsu is the name of the Joomla 4 administrator template."
```
Comment (brianteeman) [inline administrator/templates/khonsu/language/en-GB/en-GB.tpl_khonsu.sys.ini]:
```suggestion
TPL_KHONSU_XML_DESCRIPTION="Continuing the Egyptian god/goddess theme (Khepri from 1.5, Hathor from 1.6, Isis from 3.0), Khonsu is the name of a Joomla 4 administrator template."
```

File: administrator/templates/khonsu/templateDetails.xml [hunk 6]
```diff
@@ -1,12 +1,12 @@
 <?xml version="1.0" encoding="utf-8"?>
 <extension version="4.0" type="template" client="administrator">
-	<name>spring</name>
+	<name>khonsu</name>
 	<version>1.0</version>
 	<creationDate>September 2016</creationDate>
```
Comment (brianteeman) [inline administrator/templates/khonsu/templateDetails.xml]:
```suggestion
	<creationDate>November 2019</creationDate>
```

File: layouts/joomla/sidebars/submenu.php [hunk 7]
```diff
@@ -18,9 +18,9 @@
 <?php if ($displayData->displayMenu || $displayData->displayFilters) : ?>
 <div id="j-toggle-sidebar-wrapper">
 	<div id="sidebar" class="sidebar">
-		<button class="btn btn-sm btn-secondary my-2 options-menu d-md-none" type="button" data-toggle="collapse" data-target=".sidebar-nav" aria-controls="sidebar-nav" aria-expanded="false" aria-label="<?php echo Text::_('TPL_SPRING_TOGGLE_SIDEBAR'); ?>">
+		<button class="btn btn-sm btn-secondary my-2 options-menu d-md-none" type="button" data-toggle="collapse" data-target=".sidebar-nav" aria-controls="sidebar-nav" aria-expanded="false" aria-label="<?php echo Text::_('TPL_KHONSU_TOGGLE_SIDEBAR'); ?>">
```
Comment (brianteeman) [inline layouts/joomla/sidebars/submenu.php]: You should not use a template specific language string in a core non template file. See https://github.com/joomla/joomla-cms/pull/27082 for reference

File: libraries/src/Extension/ExtensionHelper.php [hunk 8]
```diff
@@ -311,7 +311,7 @@ class ExtensionHelper
 
 
 		// Core template extensions - administrator
-		array('template', 'spring', '', 1),
+		array('template', 'khonsu', '', 1),
```
Comment (brianteeman) [inline libraries/src/Extension/ExtensionHelper.php:315]: This should be **adding** khonsu and not **replacing** atum

File: libraries/src/Application/AdministratorApplication.php [hunk 9]
```diff
@@ -245,7 +245,7 @@ public function getTemplate($params = false)
 		{
 			$this->enqueueMessage(Text::_('JERROR_ALERTNOTEMPLATE'), 'error');
 			$template->params = new Registry;
-			$template->template = 'spring';
+			$template->template = 'khonsu';
```
Comment (brianteeman) [inline libraries/src/Application/AdministratorApplication.php:248]: This should be adding khonsu not replacing atum - not sure how to do the code - see `libraries\src\Application\SiteApplication.php`

Comment (ahamed) [reply libraries/src/Application/AdministratorApplication.php:248]: The scripts were written only for `Atum` template, not for multiple templates support. I think it's an another issue which you can raise.

Comment (brianteeman) [reply libraries/src/Application/AdministratorApplication.php:248]: Whatever way you look at it this code is wrong and should not be committed

Comment (marcodings) [reply libraries/src/Application/AdministratorApplication.php:248]: It was decided in production to deal with the dual/multiple template issues at a later date

**Pull request merged**